cloud_Azure/terraform/module/roles.tf

```diff
@@ -1,13 +1,17 @@
-# Provide service principal Contributor role to storage account
+# Provide service principal Contributor role to each storage account
 resource "azurerm_role_assignment" "kentic_role_contributor" {
-  scope                = azurerm_storage_account.kentik_storage_account.id
+  count = length(azurerm_storage_account.logs_storage_account)
+
+  scope                = azurerm_storage_account.logs_storage_account[count.index].id
   role_definition_name = "Contributor"
-  principal_id         = var.principal_id
+  principal_id         = azuread_service_principal.kentik_nsg_flow_exporter.object_id
 }
 
-# Provide service principal Reader role to Resource Group
+# Provide service principal Reader role to each Resource Group
 resource "azurerm_role_assignment" "kentic_role_reader" {
```

Comment:
Here too.

Reply:
Fixed on parent branch, will be merged once approved